### PR TITLE
Add asyncio

### PR DESCRIPTION
**Pull request by rudigiesler**

### CODE REVIEW

File: graas/hardware.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        '''
+        GPIO.setmode(GPIO.BOARD)
+        self.pin_state = {}
+
+        self.INPUT = GPIO.IN
+        self.OUTPUT = GPIO.OUT
+        self.HIGH = GPIO.HIGH
+        self.LOW = GPIO.LOW
+        self.PULL_UP = GPIO.PUD_UP
+        self.PULL_DOWN = GPIO.PUD_DOWN
```
Comment (hodgestar) [inline graas/hardware.py:19]: If we put these constants on the class, we can use them as default values in function definitions directly, e.g.

``` python
def input(self, pin, pull=PULL_DOWN):
    ...
```

File: graas/hardware.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+        :param int pin: The number of the pin to pulse.
+        :param value: The value to pulse. Either HIGH or LOW.
+        :param float duration: The duration to pulse for.
+
+        Returns a deferred that fires when the pulse has completed.
+        '''
+        yield self.output(pin, value)
+        yield task.deferLater(reactor, duration, self.output, pin, not value)
+        returnValue(None)
```
Comment (hodgestar) [inline graas/hardware.py:85]: Should we rather return a deferred that fires once the pulse is complete? Happy for that to be in a separate PR.

Comment (hodgestar) [inline graas/hardware.py:85]: Oh, you're using inlineCallbacks. Never mind. :)

**Comment by hodgestar** [comment]
One very minor comment, otherwise looks good.

File: graas/hardware.py (hunk 1 above)
Comment (rudigiesler) [inline graas/hardware.py:19]: I originally had them on the class, but I wanted to be able to pass in the `GPIO` class on initialization so that I could pass in a fake class for testing.

Comment (hodgestar) [inline graas/hardware.py:19]: Ah. Cool.

**Comment by hodgestar** [comment]
:+1:
